packages/cli/src/virtual-routes/routes/index.tsx

```diff
@@ -1,10 +1,178 @@
+  const layout = await context.storefront.query<{shop: Shop}>(LAYOUT_QUERY);
+  return {layout};
+}
+
+export const HYDROGEN_SHOP_ID = 'gid://shopify/Shop/55145660472';
+
+export function CatchBoundary() {
+  return <ErrorPage />;
+}
+
+export function ErrorBoundary() {
+  return <ErrorPage />;
+}
+
 export default function Index() {
+  const data = useLoaderData<typeof loader>();
+
+  const {name: shopName, id: shopId} = data.layout.shop;
+
+  const configDone = shopId !== HYDROGEN_SHOP_ID;
+
+  return (
+    <Html>
+      <Layout shopName={shopName}>
+        {configDone ? <HydrogenLogoBaseColor /> : <HydrogenLogoBaseBW />}
+        <h1>Hello, {shopName || 'Hydrogen'}</h1>
+        <p>Welcome to your new custom storefront</p>
+        <section className="Banner">
+          <h2>Configure storefront token</h2>
+          <p>
+            You're seeing this because you have not yet configured your
+            storefront token. To get started, edit <span>.env</span>. Learn more
+            about
+            {` `}
+            <a href="https://shopify.dev/custom-storefronts/hydrogen/getting-started/quickstart">
+              connecting a storefront
+            </a>
+            .
+          </p>
+        </section>
+        <ResourcesLinks />
+      </Layout>
+    </Html>
+  );
+}
+
+function ErrorPage() {
+  return (
+    <Html>
+      <Layout shopName="Hydrogen">
+        <HydrogenLogoBaseBW />
+        <h1>Hello, Hydrogen</h1>
+        <p>Welcome to your new custom storefront</p>
+        <section className="Banner ErrorBanner">
+          <h2>There’s a problem with your storefront</h2>
+          <p>
+            Check your domain and API token in your<span>.env</span> file. Read
+            the documentation on{` `}
+            <a href="">how to configure your storefront</a>.
+          </p>
+        </section>
+        <ResourcesLinks />
+      </Layout>
+    </Html>
+  );
+}
+
+function ResourcesLinks() {
+  return (
+    <>
+      <section className="Links">
+        <h2>Start building</h2>
+        <ul>
+          <li>
+            <a href="/">Collection template</a>
+          </li>
+          <li>
+            <a href="/">Product template</a>
+          </li>
+          <li>
+            <a href="/">Cart</a>
+          </li>
+        </ul>
+        <h2>Resources</h2>
+        <ul>
+          <li>
+            <a href="https://shopify.dev/custom-storefronts/hydrogen">
+              Hydrogen docs
+            </a>
+          </li>
+          <li>
+            <a href="/">Remix and project structure</a>
+          </li>
+          <li>
+            <a href="/">Data queriers and fetching</a>
+          </li>
+        </ul>
+      </section>
+    </>
+  );
+}
+
+function Layout({
+  shopName,
+  children,
+}: {
+  shopName: string;
+  children: React.ReactNode;
+}) {
   return (
     <>
-      <h2>Welcome to Hydrogen & Remix!</h2>
-      <p>
-        Create your first route in <code>routes/index.jsx</code>
-      </p>
```

Comment:
Since we are removing `routes/index.jsx` from the template, shouldn't we mention after updating `.env` that they also need to create the `routes` directory and the `index.jsx` file? cc @benjaminsehl